contrib/lkt/language/parser.py

```diff
@@ -253,10 +253,16 @@ def in_type_ref():
 
     @langkit_property()
     def designated_scope():
-        # NOTE: for the moment, every dotted expression has a typed value on
-        # the RHS, and we'll use the type to go search valid fields and methods
-        # on it. We need to add support for packages/namespaces someday.
-        return Entity.expr_type.expr_type._.type_scope
+        ref_decl = Var(Entity.referenced_decl)
+
+        # If the RHS is a type, then use the type's scope to lookup operations
```

Comment:
Why do you talk about `RHS` and not `Self`?

Reply:
You're correct, fixed